main.rb

```diff
@@ -0,0 +1,40 @@
+# bubble_sort method
+def bubble_sort(array)
+  len = array.length
+  (0...len - 1).each do |i|
+    (0...len - 1 - i).each do |j|
+      array[j], array[j + 1] = array[j + 1], array[j] if array[j] > array[j + 1]
+    end
+  end
+  array
+end
+
+# bubble_sort_by method
+def bubble_sort_by(array)
+  len = array.length - 1
+  swapped = true
+  while swapped
+    swapped = false
+    (0..len - 1).each do |i|
+      next unless yield(array[i], array[i + 1]).positive?
+
+      swap = array[i + 1]
+      array[i + 1] = array[i]
+      array[i] = swap
+      swapped = true
+    end
+    len -= 1
+  end
+  array
+end
+
+# Running the above two methods below
+
+test = [89, 4, 2, 7, 9, 4, 0, 3, -1]
+p bubble_sort(test)
+
+array_word = %w[hi hello hey hola game]
+result = bubble_sort_by(array_word) do |left, right|
+  left.length - right.length
+end
+p result
```

Comment:
**Required**: Please remove this so you only have the methods.
**Optional**: If you want to test them, you can add some RSpec tests in the right directory.